packages/node/src/utils/hot-reload.ts

```diff
@@ -40,6 +49,11 @@ export const performReload = (shouldReload: any) => {
     }
   });
   gs.__FEDERATION__.__INSTANCES__ = [];
+
+  for (const entry of entries) {
+    await __non_webpack_require__(entry);
```

Comment:
reload entrypoints again, calling async boundary and ensuring all resources are available that were cleared.